data/AlexTemplate.hs

```diff
@@ -44,18 +44,15 @@ data AlexAddr = AlexA# Addr#
 {-# INLINE alexIndexInt16OffAddr #-}
 alexIndexInt16OffAddr :: AlexAddr -> Int# -> Int#
 alexIndexInt16OffAddr (AlexA# arr) off =
-#ifdef WORDS_BIGENDIAN
-  narrow16Int# i
-  where
-        i    = word2Int# ((high `uncheckedShiftL#` 8#) `or#` low)
-        high = int2Word# (ord# (indexCharOffAddr# arr (off' +# 1#)))
-        low  = int2Word# (ord# (indexCharOffAddr# arr off'))
-        off' = off *# 2#
-#else
 #if __GLASGOW_HASKELL__ >= 901
-  GHC.Exts.int16ToInt#
+  GHC.Exts.int16ToInt# -- qualified import because it doesn't exist on older GHC's
 #endif
-    (indexInt16OffAddr# arr off)
+#ifdef WORDS_BIGENDIAN
+  (GHC.Exts.word16ToInt16# (GHC.Exts.byteSwap16# (GHC.Exts.int16ToWord16#
```

Comment:
Annoyingly [`byteSwap16#`](https://hackage.haskell.org/package/base-4.20.0.1/docs/GHC-Base.html#v:byteSwap16-35-) is `Word# - > Word#`, not `Word16# -> Word16#`, so this needs amendment.

Cf. https://gitlab.haskell.org/ghc/ghc/-/issues/22163.